src/java/htsjdk/samtools/util/IntervalList.java

```diff
@@ -233,6 +233,60 @@ else if (current.intersects(next) || current.abuts(next)) {
         return getUniqueIntervals(this, concatenateNames);
     }
 
+    /**
+     * Given a list of Intervals and a band multiple, this method will return a list of Intervals such that all of the intervals
+     * do not straddle integer multiples of that band.
+     *
+     * ex: if there is an interval (7200-9300) and the bandMultiple is 1000, the interval will be split into:
+     * (7200-7999, 8000-8999, 9000-9300)
+     * @param intervals A list of Interval
+     * @param bandMultiple integer value (> 0) to break up intervals in the list at integer multiples of
+     * @return list of intervals that are broken up
+     */
+    public static List<Interval> breakIntervalsAtBandMultiples(final List<Interval> intervals, final int bandMultiple) {
+        final List<Interval> brokenUpIntervals = new ArrayList<Interval>();
+        for (final Interval i : intervals) {
+            final int startIndex = i.getStart() / bandMultiple;
+            final int endIndex = i.getEnd() / bandMultiple;
+            if (startIndex == endIndex) {
```

Comment:
Danger! End of an interval might be less than the start (empty intervals) seems to not account for that. (also below)